pkg/controller/healthcheck/add.go

```diff
@@ -43,11 +43,7 @@ func RegisterHealthChecks(ctx context.Context, mgr manager.Manager, opts healthc
 		[]healthcheck.ConditionTypeToHealthCheck{
 			{
 				ConditionType: string(gardencorev1beta1.ShootObservabilityComponentsHealthy),
-				HealthCheck:   general.CheckManagedResource(constants.ManagedResourceNamesControllerShoot),
-			},
-			{
-				ConditionType: string(gardencorev1beta1.ShootObservabilityComponentsHealthy),
-				HealthCheck:   general.CheckManagedResource(constants.ManagedResourceNamesAgentShoot),
+				HealthCheck:   general.CheckManagedResource(constants.ManagedResourceNamesControllerSeed),
```

Comment:
You have to adapt https://github.com/gardener/gardener-extension-shoot-networking-problemdetector/blob/3a7230656da9f69a84524f96779471dc16c83019/test/integration/healthcheck/test.go#L40-L42.